cli/internal/api/templates.go

```diff
@@ -8,33 +8,113 @@ import (
 )
 
 func (c *SugaApiClient) GetTemplates(team string) ([]TemplateResponse, error) {
-	response, err := c.get(fmt.Sprintf("/api/teams/%s/templates", url.PathEscape(team)), true)
-	if err != nil {
-		return nil, err
+	// Fetch team templates and public templates in parallel
+	type templateResult struct {
+		templates []TemplateResponse
+		err       error
 	}
 
-	defer response.Body.Close()
-
-	if response.StatusCode != 200 {
-		return nil, fmt.Errorf("failed to get templates: %s", response.Status)
+	teamChan := make(chan templateResult, 1)
+	publicChan := make(chan templateResult, 1)
+
+	// Fetch team templates
+	go func() {
+		teamURL := fmt.Sprintf("/api/teams/%s/templates", url.PathEscape(team))
+
+		response, err := c.get(teamURL, true)
+		if err != nil {
+			teamChan <- templateResult{nil, err}
+			return
+		}
+		defer response.Body.Close()
+
+		if response.StatusCode != 200 {
+			teamChan <- templateResult{nil, fmt.Errorf("failed to get team templates: %s", response.Status)}
+			return
+		}
+
+		body, err := io.ReadAll(response.Body)
+		if err != nil {
+			teamChan <- templateResult{nil, err}
+			return
+		}
+
+		var templates ListTemplatesResponse
+		if err := json.Unmarshal(body, &templates); err != nil {
+			teamChan <- templateResult{nil, fmt.Errorf("failed to unmarshal team templates: %v, body: %s", err, string(body))}
+			return
+		}
+
+		teamChan <- templateResult{templates.Templates, nil}
+	}()
+
+	// Fetch public templates from suga team
+	go func() {
+		publicURL := "/api/public/templates?team_slug=suga"
+
+		response, err := c.get(publicURL, true)
+		if err != nil {
+			publicChan <- templateResult{nil, err}
+			return
+		}
+		defer response.Body.Close()
+
+		if response.StatusCode != 200 {
+			publicChan <- templateResult{nil, fmt.Errorf("failed to get public templates: %s", response.Status)}
+			return
+		}
+
+		body, err := io.ReadAll(response.Body)
+		if err != nil {
+			publicChan <- templateResult{nil, err}
+			return
+		}
+
+		var templates ListTemplatesResponse
+		if err := json.Unmarshal(body, &templates); err != nil {
+			publicChan <- templateResult{nil, fmt.Errorf("failed to unmarshal public templates: %v, body: %s", err, string(body))}
+			return
+		}
+
+		publicChan <- templateResult{templates.Templates, nil}
+	}()
+
+	// Collect results
+	teamResult := <-teamChan
+	publicResult := <-publicChan
```

Comment:
Doesn't need to be fixed as part of this PR, but the code in each of the go routines is pretty close to identical.